### PR TITLE
[853] Deprecate Link JAXB inner classes

### LINKED ISSUE #853: Deprecate JAXB usage in `Link` class.
Issue #815 is intended to make the JAXB APIs/impl optional for JAX-RS implementations, but the JAX-RS still has a dependency on JAXB from inner classes in the `Link` class.  Per discussion in #815, these inner classes should be deprecated and ultimately removed.

### PR DESCRIPTION
**Pull request by andymc12**
Deprecates the JAXB inner classes in the Link class, and modifies the example code use JSON instead of XML so as to avoid any dependencies on JAXB / Jakarta XML Binding. 

This closes #853.

### CODE REVIEW

**Review by spericas — DISMISSED** [review]
Feels good to approve this one!

**Review by mkarg — DISMISSED** [review]

**Review by chkal — APPROVED** [review]

**Comment by andymc12** [comment]
Hmm... the force push ended up pulling in several other commits unrelated to this PR.  Now that 3.0.0 is released, should I be making this change in 3.1-SNAPSHOT or master? The intent is for it to be in 3.1, but I was thinking that master was retargeted for the 3.1 release.  This is a small change, so if it makes sense for me to just open up a new/clean PR I can do that - but which branch should I base it on?  Thanks

**Comment by mkarg** [comment]
> Hmm... the force push ended up pulling in several other commits unrelated to this PR. Now that 3.0.0 is released, should I be making this change in 3.1-SNAPSHOT or master? The intent is for it to be in 3.1, but I was thinking that master was retargeted for the 3.1 release. This is a small change, so if it makes sense for me to just open up a new/clean PR I can do that - but which branch should I base it on? Thanks

Just wait a bit. I am in the middle of putting 3.1-SNAPSHOT ontop master but got stuck due to a build fail last night on JDK 14+. Once that is solved and merged, just rebase this PR's source branch ontop master *then*. This will remove the unwanted commits.

**Comment by andymc12** [comment]
@mkarg sounds good.  I did rebase the PR from 3.1-SNAPSHOT to master which removed the approvals.  So once you've got master the way we want it, I'll ask you all to re-approve.  Thanks

**Comment by mkarg** [comment]
Just waiting for your approval of https://github.com/eclipse-ee4j/jaxrs-api/pull/920. :-)

**Comment by mkarg** [comment]
@andymc12 Can you now please rebase on `master` and then force-push with the same branch name? Thanks! :-)

**Comment by andymc12** [comment]
All set - now it is rebased on master (which is targeting the 3.1 release).  @chkal @mkarg @spericas could you please re-approve?  Thanks!

**Review by chkal — APPROVED** [review]
Thanks @andymc12

**Review by mkarg — APPROVED** [review]

**Pull request merged**
